### ISSUE TITLE
Consider adding supported framework version to list output

### ISSUE THREAD

**Issue by sayedihashimi** [dotnet/templating#723]
Talking to @shanselman he's suggesting that we display the supported framework version in the output of `dotnet new --list`.

Thoughts?

**Comment by shanselman** [comment]
Current:

```
Templates                                         Short Name       Language          Tags
--------------------------------------------------------------------------------------------------------
Console Application                               console          [C#], F#, VB      Common/Console
ASP.NET Core Empty                                web              [C#]              Web/Empty
ASP.NET Core MVC Basic                            mvcbasic         [C#]              Web/MVC

```

Suggested, **very similar to Language**

```
Templates                          Short Name       Language         Version(s)       Tags                
-----------------------------------------------------------------------------------------------------
Console Application                console          [C#], F#, VB     [2.0], 1.1, 1.0  Common/Console
ASP.NET Core Empty                 web              [C#]             [2.0]            Web/Empty
ASP.NET Core MVC Basic             mvcbasic         [C#]             [2.0], 1.1       Web/MVC
```

**Comment by dansiegel** [comment]
That would be a lot nicer!

**Comment by mlorbetske** [comment]
This concerns me in the context of class library. It supports `netstandard1.0` ... `netstandard1.6`, `netstandard2.0`, `netcoreapp1.0`, `netcoreapp1.1`, `netcoreapp2.0` which will get super long & we've already had issues with the table getting too wide.

Moreover, these versions are expressed as the full value for TargetFramework - trimming to the major/minor will work as long as the TFM prefix is the same, but for netstandard/netcoreapp this will introduce ambiguity.

I'm for this change, but I think we need to find the right way to express it.

**Comment by shanselman** [comment]
Good point @mlorbetske. Are those values the same as the ones they pass in with -framework? Perhaps some abbreviation like s2.0 or a1.0 with a legend at the bottom.

**Comment by mlorbetske** [comment]
Yep, those are the same values accepted by --framework. Short forms of the TFMs are interesting, we can do that.

**Comment by donJoseLuis** [comment]
1. Adding item to specific project corresponding to a targeted delivery area
1. Once this has been done for the entire backlog, the next steps will be to prioritize within each project & output a delivery roadmap.
1. At that point, the issues will be assigned to a corresponding milestone
1. The team will then begin to deliver against the road map, contributions always welcome

**Comment by bekir-ozturk** [comment]
cc @KathleenDollard

**Comment by baronfel** [comment]
I'm in favor of this, but the length of TFMs relative to user console sizes makes me want a better table representation. The TFMs should be stacked vertically in the column, which would add vertical spacing if required. 

```
Templates                          Short Name       Language         Framework(s)       Tags                
-----------------------------------------------------------------------------------------------------
Console Application                console          [C#], F#, VB     [net7.0]         Common/Console
                                                                      net6.0
Class Library                      classlib         [C#], F#, VB     [netstandard2.0] Common/Library
                                                                      net7.0
                                                                      net6.0
```

It turns out that this is really hard to do in our current hand-rolled table implementation, though. It would be _much_ easier if we could reference Spectre.Console or something to abstract away the layout.

**Comment by marcpopMSFT** [comment]
This issue is being closed as won't fix. It has been open for an extended period without maintainer activity or community demand, and is unlikely to be addressed.

The dotnet/templating repository has been consolidated into [dotnet/sdk](https://github.com/dotnet/sdk). If this issue is still important to you, please open a new issue in [dotnet/sdk](https://github.com/dotnet/sdk/issues/new) with updated context so the team can re-evaluate.

<details>
<summary>Triage details</summary>

**Summary:** Feature request to display supported framework versions in `dotnet new --list` output

**Reasoning:** This is a 9-year-old feature request to add framework version info to `dotnet new --list` output. It has zero upvotes, no activity since 2023, and the last substantive discussion was in 2022. While it involved maintainers (shanselman, baronfel), it was never prioritized and the CLI output has evolved significantly since 2017. The stale age and lack of community demand suggest this is unlikely to be addressed.

</details>
